Mage.Sets/src/mage/cards/i/IllicitMasquerade.java

```diff
@@ -0,0 +1,114 @@
+package mage.cards.i;
+
+import java.util.UUID;
+
+import mage.MageItem;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbility;
+import mage.abilities.common.DiesCreatureTriggeredAbility;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.effects.common.ExileTargetEffect;
+import mage.abilities.effects.common.ReturnFromGraveyardToBattlefieldTargetEffect;
+import mage.abilities.effects.common.counter.AddCountersAllEffect;
+import mage.abilities.keyword.FlashAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.TargetController;
+import mage.counters.CounterType;
+import mage.filter.FilterCard;
+import mage.filter.FilterPermanent;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterCreatureCard;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.ObjectSourcePlayer;
+import mage.filter.predicate.ObjectSourcePlayerPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.target.common.TargetCardInYourGraveyard;
+import mage.target.targetpointer.FirstTargetPointer;
+
+/**
+ *
+ * @author DominionSpy
+ */
+public final class IllicitMasquerade extends CardImpl {
+
+    public IllicitMasquerade(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{3}{B}");
+
+        // Flash
+        this.addAbility(FlashAbility.getInstance());
+
+        // When Illicit Masquerade enters the battlefield, put an impostor counter on each creature you control.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new AddCountersAllEffect(
+                CounterType.IMPOSTOR.createInstance(), StaticFilters.FILTER_CONTROLLED_CREATURE)));
+
+        // Whenever a creature you control with an impostor counter on it dies, exile it. Return up to one other target creature card from your graveyard to the battlefield.
+        this.addAbility(new IllicitMasquerageAbility());
+    }
+
+    private IllicitMasquerade(final IllicitMasquerade card) {
+        super(card);
+    }
+
+    @Override
+    public IllicitMasquerade copy() {
+        return new IllicitMasquerade(this);
+    }
+}
+
+class IllicitMasquerageAbility extends DiesCreatureTriggeredAbility {
```

Comment:
Look at similar card like [Body Snatcher](https://scryfall.com/card/dmr/75/body-snatcher) - it uses standard effects without new classes. Is it possible to use same way here?

Reply:
it's due to the two effects both needing targets here - one fixed target set by the trigger, the other first target pointer. (Body Snatcher has exile source effect so only one target needed.)